cytodynamics-test-container/src/test/java/com/linkedin/cytodynamics/test/TestDynamicLoad.java

```diff
@@ -245,6 +271,62 @@ public void testLoadAandB() {
     assertEquals(implementationB.getValue(), "B");
   }
 
+  @Test(description = "Given that there are multiple parent relationships, and all can load a class, the class should "
+      + "be loaded from the first parent relationship")
+  public void testMultipleParentsLoadFromFirstParent() throws MalformedURLException {
+    URL testApiJarURL = getTestJarUri("api").toURL();
+    ClassLoader parentClassLoaderA = new URLClassLoader(new URL[]{testApiJarURL, getTestJarUri("a").toURL()}, null);
+    ClassLoader parentClassLoaderB = new URLClassLoader(new URL[]{testApiJarURL, getTestJarUri("b").toURL()}, null);
+    Loader loader = LoaderBuilder
+        .anIsolatingLoader()
+        .withOriginRestriction(OriginRestriction.allowByDefault())
+        // will load class from parent, so don't need any classpath
+        .withClasspath(Collections.emptyList())
+        .addParentRelationship(ParentRelationshipBuilder.builder()
+            .withParentClassLoader(parentClassLoaderA)
+            // using NONE so that we can load the class from the parent
+            .withIsolationLevel(IsolationLevel.NONE)
+            .addWhitelistedClassPattern("java.*")
+            .build())
+        .addParentRelationship(ParentRelationshipBuilder.builder()
+            .withParentClassLoader(parentClassLoaderB)
+            // using NONE so that we can load the class from the parent
+            .withIsolationLevel(IsolationLevel.NONE)
+            .addWhitelistedClassPattern("java.*")
+            .build())
+        .build();
+    Class<?> clazz = loader.loadClass(Object.class, TestInterfaceImpl.class.getName());
+    assertEquals(clazz.getClassLoader(), parentClassLoaderA);
+  }
+
+  @Test(description = "Given that there are multiple parent relationships, but only the last parent has a class, the "
+      + "class should be loaded by the last parent")
+  public void testMultipleParentsLoadFromLastParent() throws MalformedURLException {
```

Comment:
It is good to have these two tests, but considering that having multiple parents may be quite gnarly, can we consider adding more test scenarios?

For example, the Samza consumer scenario described in the commit message does not appear to be fully covered here, since there is nothing that's ever loaded from A. In your example scenario, IIUC, we would want to try loading `@Api` classes from A. Can we do that in the test (either this one or another one)?

Reply:
I'll look into adding more coverage.

Reply:
I added a test which better simulates the example mentioned in the description. Thanks for the suggestion.
I couldn't come up with other scenarios that aren't already covered by the 3 new tests though. Did you have any other cases?